python/cudf/cudf/core/dataframe.py

```diff
@@ -6454,12 +6473,36 @@ def _apply_support_method(self, method, axis=0, *args, **kwargs):
                 "support `bool_only`."
                 raise NotImplementedError(msg)
 
-            prepared = self._prepare_for_rowwise_op()
+            prepared = self._prepare_for_rowwise_op(method)
+            if not skipna and any(col.nullable for col in prepared._columns):
+                mask = cudf.DataFrame(
+                    {
+                        name: prepared._data[name]._get_mask_as_column()
+                        if prepared._data[name].nullable
+                        else column.full(len(prepared._data[name]), True)
+                        for name in prepared._data.names
+                    }
+                )
+                mask = mask.all(axis=1)
+            else:
+                mask = None
+
+            prepared = prepared.astype("float64")
+            prepared = prepared.fillna(np.nan)
```

Comment:
Shouldn't we only cast to `float64` if there's nulls or non-homogenous numerical types? I.e. if all of the columns are non-nullable int32 should we keep it as int32 and not run the `fillna`?

Reply:
What is Pandas behavior here?

Reply:
> What is Pandas behavior here?

```python
>>> df = pd.DataFrame({'a':[1, 2, 3, 4], 'b':pd.Series([10, 11, 12, None], dtype=pd.Int64Dtype())})
>>> df
   a     b
0  1    10
1  2    11
2  3    12
3  4  <NA>
>>> df.min()
a     1
b    10
dtype: int64
>>> df.min(axis=1)
0    1.0
1    2.0
2    3.0
3    4.0
dtype: float64
```

I guess I should be type-casting only when there are nulls. We might not have to check for non-homogenous types for type-casting, since cupy is up-casting:

```python
>>> import cudf
>>> df = cudf.DataFrame({"a":[1, 2, 3, 4], "b":[0.32, 0.23, 0.2, 0.2]})
>>> import cupy
>>> cupy.asarray(df.as_gpu_matrix())
array([[1.  , 0.32],
       [2.  , 0.23],
       [3.  , 0.2 ],
       [4.  , 0.2 ]])
>>> cupy.asarray(df.as_gpu_matrix()).dtype
dtype('float64')
>>> df = cudf.DataFrame({"a":[1, 2, 3, 4], "b":[10, 11, 12, 13]})
>>> cupy.asarray(df.as_gpu_matrix()).dtype
dtype('int64')
```

Reply:
I think we want to rely on that type-casting behavior, i.e. if we have `int8`, `int16`, and `int32` columns we can safely use `int32` to represent the results. If we have `float32` and `int32`, we should use `int64`.

Reply:
> . If we have `float32` and `int32`, we should use `int64`.

I think this is `float64` 

```python
>>> np.find_common_type(["float32", "int32"], [])
dtype('float64')
```



> I think we want to rely on that type-casting behavior,

Yes, made a change to rely on that type-casting behavior and only changing the dtype in case of nullable columns

Reply:
> I think this is `float64`

Sorry, typo! 😅